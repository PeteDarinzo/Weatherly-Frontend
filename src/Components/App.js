import './App.css';
import React, { useState, useEffect } from "react";
import { useHistory } from 'react-router-dom';
import NavBar from "./NavBar";
import Routes from "./Routes";
import WeatherlyApi from './Api';
import jwt_decode from "jwt-decode";
import { sendMovieToAPI } from '../Actions/actions';
import { useDispatch } from "react-redux";

function App() {

  const history = useHistory();
  const dispatch = useDispatch();

  const [isLoading, setIsLoading] = useState(true);
  const [userToken, setUserToken] = useState("");
  const [userCredentials, setUserCredentials] = useState("");
  const [searchResults, setSearchResults] = useState({});

  useEffect(() => {
    if (localStorage.getItem("weatherlyToken")) {
      const token = JSON.parse(localStorage.getItem("weatherlyToken"));
      // WeatherlyApi.token = token; // set token for future reqeusts
      setUserToken(token);
    }
  }, []);

  useEffect(() => {
    if (userToken) {
      const decoded = jwt_decode(userToken);
      const username = decoded.username;
      setUserCredentials(username);
    }
  }, [userToken]);

  /** Search OMDB for movies */
  async function getMovies(title) {
    const res = await WeatherlyApi.searchMovies(title);
    setSearchResults(res);
  }


  /** Search OMDB for movies */
  // async function getMovies(title) {
  //   const res = await axios.get(`${API_URL}/movies/title/${title}`);
  //   setSearchResults(res.data);
  // }

  /** Register a new users */
  async function register(user) {
    const userObj = {
      username: user.username,
      password: user.password,
      zipCode: user.zipCode
    }
    // setIsLoading(true);
    const token = await WeatherlyApi.register(userObj);
    // const res = await axios.post(`${API_URL}/auth/register`, userObj);
    // const token = res.data.token;
    localStorage.setItem("weatherlyToken", JSON.stringify(token));
    setUserToken(token);
    history.push("/");
  }

  async function login(user) {
    const userObj = {
      username: user.username,
      password: user.password,
    }
    const token = await WeatherlyApi.getToken(userObj);
    // const res = await axios.post(`${API_URL}/auth/token`, userObj);
    // const token = res.data.token;
    localStorage.setItem("weatherlyToken", JSON.stringify(token));
    setUserToken(token);
    history.push("/");
  }

  function logout() {
    localStorage.removeItem("weatherlyToken");
    setUserToken("");
  }

  function saveMovie(movie) {
    dispatch(sendMovieToAPI(movie, userCredentials));
  }

  return (
    <div>
      <NavBar loggedIn={userToken} logout={logout} />
      <Routes
        getMovies={getMovies}
        searchResults={searchResults}
        register={register}
        login={login}
        loggedIn={userToken}
        saveMovie={saveMovie}
        username={userCredentials}
      />
    </div>
  );
}

export default App;
