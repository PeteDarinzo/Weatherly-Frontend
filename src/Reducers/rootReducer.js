
import { combineReducers } from "redux";
import titlesReducer from "./titles";
import moviesReducer from "./movies";
import forecastReducer from "./forecast";
import userReducer from "./user";
import snackbarReducer from "./snackbar";

const rootReducer = combineReducers({
  titles: titlesReducer,
  movies: moviesReducer,
  forecast: forecastReducer,
  user: userReducer,
  snackbar: snackbarReducer
});

export default rootReducer;