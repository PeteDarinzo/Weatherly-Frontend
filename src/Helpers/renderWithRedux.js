import { render } from '@testing-library/react';
import { createStore, applyMiddleware } from "redux";
import { Provider } from 'react-redux';
import rootReducer from "../Reducers/rootReducer";
import thunk from "redux-thunk"


export default function renderWithRedux(component, { store = createStore(rootReducer, applyMiddleware(thunk)) } = {}
) {
  return {
    ...render(
      <Provider store={store}>
        {component}
      </Provider>)
  }
}