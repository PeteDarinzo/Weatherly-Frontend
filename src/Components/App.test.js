import React from "react";
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import { Provider } from 'react-redux';
import App from './App';
import { createStore } from "redux";
import rootReducer from "../Reducers/rootReducer";
import renderWithRedux from "../Helpers/renderWithRedux";

afterEach(cleanup);


// function renderWithRedux(component, { initialState, store = createStore(rootReducer, initialState) } = {}
// ) {
//   return {
//     ...render(
//       <Provider store={store}>
//         {component}
//       </Provider>)
//   }
// }

test('renders App component with redux', () => {
  renderWithRedux(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
});


it("matches the home snapshot", () => {
  const { asFragment } = renderWithRedux(
    <MemoryRouter>
      <App />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
