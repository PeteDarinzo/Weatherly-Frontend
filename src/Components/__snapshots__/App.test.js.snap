// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the home snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1rjyvvz-MuiPaper-root"
  >
    <div
      class="MuiBox-root css-i9gxme"
    >
      <header
        class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-1k0i2s9-MuiPaper-root-MuiAppBar-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-1t29gy6-MuiToolbar-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h3 css-snohim-MuiTypography-root"
          >
            Weatherly
          </div>
          <div>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1fitntr-MuiButtonBase-root-MuiButton-root"
              href="/signup"
              tabindex="0"
            >
              Signup
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
            <a
              class="MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-s1oay0-MuiButtonBase-root-MuiButton-root"
              href="/login"
              tabindex="0"
            >
              Login
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </header>
    </div>
    <div />
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h2 css-qq3owz-MuiTypography-root"
          >
            WELCOME TO WEATHERLY
          </h2>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6 css-1osj8n2-MuiGrid-root"
        >
          <h5
            class="MuiTypography-root MuiTypography-h5 css-1jcyw1h-MuiTypography-root"
          >
            What is it?
          </h5>
          <p
            class="MuiTypography-root MuiTypography-body1 css-zpphrz-MuiTypography-root"
          >
            Your time is valuable.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-zpphrz-MuiTypography-root"
          >
            Don't scroll endlessly looking for something to watch on a sunny day.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-zpphrz-MuiTypography-root"
          >
            Keep your "to-watch" list close at hand.
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-zpphrz-MuiTypography-root"
          >
            Let the app find the next inclement weather day for you.
          </p>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-0"
          >
            <p
              class="MuiTypography-root MuiTypography-body1 css-fksj50-MuiTypography-root"
            >
              New to Weatherly?
              <a
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-syebu3-MuiButtonBase-root-MuiButton-root"
                href="/signup"
                tabindex="0"
              >
                Signup
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </p>
            <p
              class="MuiTypography-root MuiTypography-body1 css-fksj50-MuiTypography-root"
            >
              Already a user?
              <a
                class="MuiButton-root MuiButton-outlined MuiButton-outlinedSecondary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-4cljzk-MuiButtonBase-root-MuiButton-root"
                href="/login"
                tabindex="0"
              >
                Login
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
